src/store/channels/saga.ts

```diff
@@ -110,6 +110,18 @@ export function* onFavoriteRoom(action) {
   yield call(addRoomToFavorites, roomId);
 }
 
+export function* onUnfavoriteRoom(action) {
+  const { roomId } = action.payload;
+
+  const channel = yield select(rawChannelSelector(roomId));
+
+  if (!channel) {
+    return;
+  }
+
+  yield call(removeRoomFromFavorites, roomId);
```

Comment:
We should probably wrap these in try/catch statements. What if Matrix fails to execute this...what should we tell the user?

Reply:
Follow up was going to include some form of basic error state for either of these saga functions failing. For this PR I have now added try/catch statements, simply logging error. Shall implement something for the user as a follow up